homeassistant/components/notify/pushbullet.py

```diff
@@ -24,27 +27,81 @@ def get_service(hass, config):
         return None
 
     try:
-        return PushBulletNotificationService(config[CONF_API_KEY])
-
+        pushbullet = PushBullet(config[CONF_API_KEY])
     except InvalidKeyError:
         _LOGGER.error(
             "Wrong API key supplied. "
             "Get it at https://www.pushbullet.com/account")
         return None
 
+    return PushBulletNotificationService(pushbullet)
+
 
 # pylint: disable=too-few-public-methods
 class PushBulletNotificationService(BaseNotificationService):
     """ Implements notification service for Pushbullet. """
 
-    def __init__(self, api_key):
-        from pushbullet import Pushbullet
+    def __init__(self, pb):
+        self.pushbullet = pb
+        self.pbtargets = {}
+        self.refresh()
 
-        self.pushbullet = Pushbullet(api_key)
+    def refresh(self):
+        '''
+        Refresh devices, contacts, channels, etc
 
-    def send_message(self, message="", **kwargs):
-        """ Send a message to a user. """
+        pbtargets stores all targets available from this pushbullet instance
+        into a dict. These are PB objects!. It sacrifices a bit of memory
+        for faster processing at send_message
+        '''
+        self.pushbullet.refresh()
+        self.pbtargets = {
+            'devices':
+                {tgt.nickname: tgt for tgt in self.pushbullet.devices},
+            'contacts':
+                {tgt.email: tgt for tgt in self.pushbullet.contacts},
+            'channels':
+                {tgt.channel_tag: tgt for tgt in self.pushbullet.channels},
+        }
 
+    def send_message(self, message=None, **kwargs):
+        """
+        Send a message to a specified target.
+        If no target specified, a 'normal' push will be sent to all devices
+        linked to the PB account.
+        """
+        targets = kwargs.get(ATTR_TARGET)
+        # Disabeling title
         title = kwargs.get(ATTR_TITLE)
+        title = None
+
+        if targets:
+            # Make list if not so
+            if not isinstance(targets, list):
+                targets = [targets]
+
+            # Main loop, Process all targets specified
+            for ttype, tname in [target.split('.', 1) for target in targets]:
+                if ttype == 'device' and not tname:
```

Comment:
Setting target to `device` will result in a `ValueError`:

```python
>>> ttype, tname = 'device'.split('.', 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: need more than 1 value to unpack
```

Instead, this will work:
```python
for target in targets:
    if target == 'device':
      …
      continue

    ttype, tname = target.split('.', 1)
```